Khan/mathquill
# Add types for `cursor()` and `controller()` in mathquill.d.ts

## Summary:

Currently, the type for `cursor()` and `controller()` on `Mathfield` are `any`. We want them to be properly typed.

In this PR:
- Add types for `controller()` and `cursor()` in build/mathquill.d.ts.
- Remove these in src/mathquill.d.ts since they're not being used here, and they'd mess up types in other places.

Issue: https://khanacademy.atlassian.net/browse/LC-1659

## Test plan:

- Copy all the new contents of [build/mathquill.d.ts](https://github.com/Khan/mathquill/blob/main/build/mathquill.d.ts)
- Paste them in khan/perseus/node_modules/mathquill/build/mathquill.d.ts (local file)
- Run `yarn typecheck` in the Perseus repo
- Confirm there are no type errors

**Testing note:**
This requires changing line 24 in [khan/perseus/packages/math-input/src/components/input/__tests__/test-math-wrapper.ts](https://github.com/Khan/perseus/blob/e853bbbff4f9ee0886801ad48e9af5d7e1e3d226/packages/math-input/src/components/input/__tests__/test-math-wrapper.ts#L24) from
```
return selection.ends[-1][-1] === 0 && selection.ends[1][1] === 0;
```
to 
```
return (
    selection.getEnd([-1][-1]) === 0 &&
    selection.getEnd([1][1]) === 0
);
```
